flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/taskmanager/TaskManagerDetailsHandler.java

```diff
@@ -117,7 +118,8 @@ public TaskManagerDetailsHandler(
 			);
 	}
 
-	private static TaskManagerMetricsInfo createTaskManagerMetricsInfo(MetricStore.TaskManagerMetricStore tmMetrics) {
+	@VisibleForTesting
+	static TaskManagerMetricsInfo createTaskManagerMetricsInfo(MetricStore.TaskManagerMetricStore tmMetrics) {
```

Comment:
I would avoid `VisibleForTesting` shortcuts
and test directly `TaskManagerDetailsHandler::handleRequest`
like in `TaskManagerLogListHandlerTest`.